bot/cogs/stats.py

```diff
@@ -0,0 +1,93 @@
+from discord import Member, Message, Status
+from discord.ext.commands import Bot, Cog, Context
+
+from bot.constants import Channels, Guild
+
+
+CHANNEL_NAME_OVERRIDES = {
+    Channels.off_topic_0: "off_topic_0",
+    Channels.off_topic_1: "off_topic_1",
+    Channels.off_topic_2: "off_topic_2",
+    Channels.staff_lounge: "staff_lounge"
+}
+
+
+class Stats(Cog):
+    """A cog which provides a way to hook onto Discord events and forward to stats."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @Cog.listener()
+    async def on_message(self, message: Message) -> None:
+        """Report message events in the server to statsd."""
+        if message.guild is None:
+            return
+
+        if message.guild.id != Guild.id:
+            return
+
+        reformatted_name = message.channel.name.replace('-', '_')
```

Comment:
Will Unicode be an issue. In any case, stripping emojis from help channels would be a good idea.

Reply:
Regarding help channels, the cog has a function you can borrow to strip emojis.

Reply:
Stripped to alphanumeric